src/primitives/tweedlesig.rs

```diff
@@ -0,0 +1,36 @@
+//! TODO
+//!
+//! - If we're using the other curve, then this won't be an instantiation of RedDSA
+//! - let's use "Verification" and "Signing" key terminology
+
+use std::marker::PhantomData;
+
+pub trait SigType: private::Sealed {}
+
+pub enum SpendAuth {}
+impl SigType for SpendAuth {}
+
+pub enum Binding {}
+impl SigType for Binding {}
+
+pub struct SigningKey<T: SigType> {
+    _t: PhantomData<T>,
+}
+
+pub struct VerificationKey<T: SigType> {
```

Comment:
```suggestion
pub struct ValidationKey<T: SigType> {
```

Reply:
Hmm, that's inconsistent with pretty much every signature library I can think of. I've seen `PublicKey` (which I dislike) and (variants of) `VerificationKey`, but never `ValidatingKey`.